src/pages/api/auth/[...nextauth].ts

```diff
@@ -39,8 +40,14 @@ export function getAuthOptions(req: IncomingMessage): NextAuthOptions {
           }
 
           await siwe.verify({ signature: credentials?.signature ?? "" });
+          const keypair = genKeyPair();
```

Comment:
Doesn't this generate a random keypair?

> Signature generation is a deterministic process here because it depends on your connected wallet

How is this true? I think I'm missing something.

Reply:
Key pair is generated only once, when user signs the message.
maci-cli doesn't have other methods to generate key pair.

Reply:
We are generating random keypair here:
https://github.com/privacy-scaling-explorations/maci/blob/e2e10313abb22d6878983e9324788c734e9721ce/cli/ts/commands/genKeyPair.ts#L13-L14

That's why I ask the question in dc channel: should we genKeyPair based on something related to the user's wallet? such as signature (sign on something)?

Reply:
import { Keypair } from "maci-domainobjs"
const keypair = new Keypair(signature);

Reply:
That's what I would like it to be 👍

Reply:
@ctrlc03 seems it won't create the same keypair from signature because it's always different due to nonce and issuedAt. Not sure how to tweak this signing process but these two field must be removed from the message object to get the same key pair every time.

Reply:
Ok, it's good now, issuesAt doesn't affect signature and nonce is always the same.

Reply:
> Ok, it's good now, issuesAt doesn't affect signature and nonce is always the same.

How is nonce is always the same?

Reply:
@samajammin it's the same approach as with csrf token so it's unique for all the users. That's why we always have the same keypair and we don't need to store it.